pkg/cloudencrypt/metadata.go

```diff
@@ -0,0 +1,14 @@
+package cloudencrypt
+
+type MetadataKV struct {
+	Key   string
+	Value string
+}
+
+func buildMetadataMap(metadata ...MetadataKV) map[string]string {
+	out := make(map[string]string)
+	for _, kv := range metadata {
+		out[kv.Key] = kv.Value
+	}
+	return out
+}
```

Comment:
I don't like this module.
It is just slice -> map conversion. The structure has no methods on that and the only method is to create the map. Why can't we use `metadataKV map[string]string` as default value? Instead we pass into function this `...MetadataKV` which can be absent.
Try to rework it to plain types with simple checks instead of wrapping and converting it each pass of function

Reply:
See for example https://github.com/prometheus/client_golang/blob/main/prometheus/labels.go#L33